master/security/kubernetes-default-deny.md

```diff
@@ -0,0 +1,105 @@
+
+For compatibility with Kubernetes, **Calico network policy** enforcement follows the standard convention for Kubernetes pods:
+- If no network policies apply to a pod, then all traffic to/from that pod is allowed.
+- If one or more network policies apply to a pod containing ingress rules, then only the ingress traffic specifically allowed by those policies is allowed, and all other traffic to/from the pod is denied.
+- If one or more network policies apply to a pod containing egress rules, then only the egress traffic specifically allowed by those policies is allowed, and all other traffic to/from the pod is denied.
+
+For other endpoint types (VMs, host interfaces), **Calico global network policy is default deny**. Only traffic specifically allowed by network policy is allowed, even if no network policies apply to the endpoint.
+
+#### Secure resources everywhere
+
+We recommend creating default deny policy using **Calico global network policy**. It applies to both workloads (VMs and containers) and hosts (computers that run the hypervisor for VMs, or container runtime for containers). Using a Calico global network policy supports a conservative security stance for protecting resources.
+
+#### Best practice: implicit default deny policy
+
+We recommend that you create a "plain vanilla" default deny Calico network policy that does not include other deny rules. Why? The absence of other deny rules ensures that the default deny policy is implicitly "first/always". 
+
+If you add other deny rules, you must also add an **order number** to ensure the policy always remains in affect (by virtue of being the highest order number across all of your ordered policies). An **implicit default deny policy** is not only easier, but avoids mistakes. 
+
+### How to
+
+Although you can use any of the following policies to create default deny for Kubernetes pods, we recommend using the Calico policy, non-namespaced.
+
+- [Create default deny-all traffic Calico policy, non-namespaced](#create-default-deny-all-traffic-calico-policy-non-namespaced)
+- [Create default deny-all traffic Calico policy, namespaced](#create-default-deny-all-traffic-calico-policy-namespaced)
+- [Create default deny all traffic Kubernetes policy, namespaced](#create-default-deny-all-traffic-Kubernetes-policy-namespaced)
+
+#### Create default deny-all traffic Calico policy, non-namespaced
+
+In the following example, we specify a default deny **GlobalNetworkPolicy** (recommended) that denies all traffic to workload endpoint resources in all namespaces, and to host endpoint resources. It is an implicit default deny policy without rules; it will always be in effect, regardless of the ordering scheme that you may specify in normal policies. 
```

Comment:
If we're going to say "(recommended)" here I think we should take a sentence to explain _why_ it is recommended.

We probably can't say across the board in the general case that this is the recommended approach, though. 

We can say "This approach provides the most-secure baseline, requiring that all traffic to and from pods in the cluster is explicitly allowed by a network policy"

Reply:
>  that denies all traffic to workload endpoint resources in all namespace

And, related to that subtle point above, this policy actually doesn't deny any traffic. It only implement a default-deny if the traffic isn't allowed. 

Said another way - this policy doesn't deny any traffic, it simply doesn't allow any traffic.

Reply:
Fixed.